src/users_db.rs

```diff
@@ -150,9 +163,45 @@ impl UsersDb {
         )
     }
 
-    pub fn read(&self) -> rusqlite::Result<Vec<User>> {
-        let mut stmt = try!(self.connection.prepare("SELECT * FROM users"));
-        let rows = try!(stmt.query(&[]));
+    pub fn read(&self, filter: ReadFilter) -> rusqlite::Result<Vec<User>> {
+        let mut stmt = try!(
+            self.connection.prepare("SELECT * FROM users")
+        );
+
+        let rows = match filter {
+            ReadFilter::All => {
+                try!(stmt.query(&[]))
+            },
+            ReadFilter::Id(id) => {
+                stmt = try!(
+                    self.connection.prepare("SELECT * FROM users WHERE id=$1")
+                );
+                try!(stmt.query(&[&id]))
+            },
+            ReadFilter::Name(name) => {
+                stmt = try!(
+                    self.connection.prepare("SELECT * FROM users WHERE name=$1")
+                );
+                try!(stmt.query(&[&escape(&name)]))
+            },
+            ReadFilter::Email(email) => {
+                stmt = try!(
+                    self.connection.prepare("SELECT * FROM users WHERE email=$1")
+                );
+                try!(stmt.query(&[&escape(&email)]))
+            },
+            ReadFilter::Credentials(user, password) => {
+                let mut md5 = Md5::new();
+                md5.input_str(&escape(&password));
```

Comment:
maybe we don't need to scape the password here since we are going to have a md5 out of it right? Anyway, not bad imo, to leave it.